File: VECTOAux/VectoAuxiliaries/Electrics/AltUserInput.vb
Option Strict On


Namespace Electrics

'Used by the Combined Alternator Form/Classes to accept user input for the combined alternators efficiency
'At different Current Demands
  Public class AltUserInput
  
    Public Amps As Single
    Public Eff As Single 
  
    'Constructor
    Sub new( amps As Single , eff As single)

   Me.Amps=amps
   Me.Eff = eff

  End Sub
  
   'Equality
    Public Function IsEqual(  other As AltUserInput, Optional rounding As Integer=7) As Boolean

    Return Math.round(Me.Amps,rounding)= Math.Round(other.Amps,rounding) AndAlso _
        Math.Round(Me.Eff,rounding) = Math.Round(other.eff,rounding)

    End Function
  
  
  End class

End Namespace

File: VectoCore/VectoCore/Models/BusAuxiliaries/Interfaces/DownstreamModules/Electrics/AltUserInput.cs
using System;
using TUGraz.VectoCommon.Utils;

namespace TUGraz.VectoCore.Models.BusAuxiliaries.Interfaces.DownstreamModules.Electrics
{
	// Used by the Combined Alternator Form/Classes to accept user input for the combined alternators efficiency
	// At different Current Demands
	public class AltUserInput<T> where T: SI
	{
		public T Amps;
		public double Eff;

		// Constructor
		public AltUserInput(T amps, double eff)
		{
			Amps = amps;
			Eff = eff;
		}

		// Equality
		public bool IsEqual(AltUserInput<T> other, int rounding = 7)
		{
			return Amps != null && other != null && Amps.IsEqual(other.Amps, Math.Pow(10, -rounding).SI<Ampere>()) &&
					Eff.IsEqual(other.Eff, Math.Pow(10, -rounding));
		}
	}
}
